library/ssl_ciphersuites.c

```diff
@@ -2209,15 +2210,30 @@ const int *mbedtls_ssl_list_ciphersuites( void )
 #else
             if( mbedtls_ssl_ciphersuite_from_id( *p ) != NULL )
 #endif
+            {
                 *(q++) = *p;
+                supported_count++;
+            }
         }
         *q = 0;
 
         supported_init = 1;
     }
+}
+
+const int *mbedtls_ssl_list_ciphersuites( void )
+{
+    mbedtls_ssl_ciphersuites_init();
 
     return( supported_ciphersuites );
 }
+
+int mbedtls_ssl_ciphersuites_count( void )
```

Comment:
This function should e implemented also if `MBEDTLS_SSL_CIPHERSUITES` is defined